Makefile

```diff
@@ -18,18 +22,60 @@ clean: ## delete most git-ignored files
 	find . -name '*.pyc' -exec rm -f {} +
 	find . -name '*.pyo' -exec rm -f {} +
 	find . -name '*~' -exec rm -f {} +
+	coverage erase
+	rm -fr *.egg-info
 
 requirements: ## install environment requirements
 	pip install -r requirements/base.txt
 
+install-automation-reqs: ## install tox
+	$(TOX) pip install -r requirements/tox.txt
+
+tox-test-reqs: install-automation-reqs
+	$(TOX) pip install -r requirements/test.txt --exists-action w
+
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -qr requirements/pip-tools.txt
 	# Make sure to compile files after any other files they include!
 	$(PIP_COMPILE) -o requirements/pip-tools.txt requirements/pip-tools.in
 	$(PIP_COMPILE) -o requirements/base.txt requirements/base.in
 	$(PIP_COMPILE) -o requirements/test.txt requirements/test.in
+	$(PIP_COMPILE) -o requirements/tox.txt requirements/tox.in
+
+quality: clean tox-test-reqs## check coding style with pycodestyle and pylint
+	$(TOX) pycodestyle ./eox_nelp
+	$(TOX) pylint ./eox_nelp --rcfile=./setup.cfg
+	$(TOX) isort --check-only --diff ./eox_nelp
+
+python-test: clean tox-test-reqs## Run test suite.
+	$(TOX) coverage run --source ./eox_nelp manage.py test
+	$(TOX) coverage report -m --fail-under=71
+
+complexity: clean tox-test-reqs## Run complexity suite with flake8.
+	$(TOX) flake8  --max-complexity 10 eox_nelp
+
+run-tests: python-test quality complexity
+
+##-----------------------------Translations section-------------------------
+extract-translations: ## extract strings to be translated, outputting .mo files
+	./manage.py makemessages -l ar   -i manage -i setup -i "venv/*"
+
+compile-translations: ## compile translation files, outputting .po files for each supported language
+	cd eox_nelp && ../manage.py compilemessages
+
+detect-changed-source-translations:
+	cd eox_nelp && i18n_tool changed
+
+pull-translations: ## pull translations from Transifex
+	tx pull -af --mode reviewed
+
+push-translations: ## push source translation files (.po) from Transifex
```

Comment:
same comment as above

Reply:
The tx command is if in the future we are going to add transifex cli and start using it. by the way, seems I should remove them right?

Reply:
I think we won't include this in transifex , that's why I think is not necessary, however we will work a lot in this pluginsso there is a chance to use it on the other hand we will start to implement mfes so this will be just backend, whatever, you decide, is ok